### PR TITLE
decompile pokedex_02074944.c

### PR DESCRIPTION
**Pull request by PikalaxALT**
might rename to something like "party_battle_rules_check"

### CODE REVIEW

File: src/battle_regulation.c [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+#include "constants/items.h"
+
+#include "dex_mon_measures.h"
+#include "party.h"
+#include "pokemon.h"
+#include "unk_020290B4.h"
+
+static BOOL sub_02074C50(u16 *a0, u16 *a1, u16 *a2, int a3, int a4, int a5, int a6);
+
+PokedexData *sub_02074944(HeapID heapId) {
```
Comment (adrienntindall) [inline src/battle_regulation.c:14]: Can be named to something like Pokedex_CreateAndLoad, Pokedex_Load, etc, whatever you see fit

File: src/battle_regulation.c [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    PokedexData *pokedex = PokedexData_Create(heapId);
+    PokedexData_LoadAll(pokedex, 0, heapId);
+    return pokedex;
+}
+
+void sub_0207495C(PokedexData *pokedex) {
```
Comment (adrienntindall) [inline src/battle_regulation.c:20]: Pokedex_Free

File: src/battle_regulation.c [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+
+#include "dex_mon_measures.h"
+#include "party.h"
+#include "pokemon.h"
+#include "unk_020290B4.h"
+
+static BOOL sub_02074C50(u16 *a0, u16 *a1, u16 *a2, int a3, int a4, int a5, int a6);
+
+PokedexData *sub_02074944(HeapID heapId) {
+    PokedexData *pokedex = PokedexData_Create(heapId);
```
Comment (adrienntindall) [inline src/battle_regulation.c:15]: any reason why the struct is called PokedexData instead of just Pokedex?

File: src/battle_regulation.c [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+    PokedexData_Delete(pokedex);
+}
+
+BOOL sub_0207496C(LinkBattleRuleset *ruleset, Pokemon *pokemon, PokedexData *pokedex) {
+    u16 species = GetMonData(pokemon, MON_DATA_SPECIES, NULL);
+    if (ruleset == NULL) {
+        return TRUE;
+    }
+
+    int rule = LinkBattleRuleset_GetRuleValue(ruleset, 2);
```
Comment (adrienntindall) [inline src/battle_regulation.c:31]: I'm guessing the one int is required to match? otherwise having descriptive variables like level would be more ideal

Comment (adrienntindall) [reply src/battle_regulation.c:31]: Also you can change 2 to something like RULEVAL_LEVEL or whatever you deem appropriate, etc for all other known consts

File: src/battle_regulation.c [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+    if (rule == 0) {
+        if (MonCheckFrontierIneligibility(pokemon)) {
+            return FALSE;
+        }
+    }
+
+    return TRUE;
+}
+
+int sub_02074A6C(LinkBattleRuleset *a0, Party *party, PokedexData *pokedex, u8 *selectedOrder) {
```
Comment (adrienntindall) [inline src/battle_regulation.c:89]: this looks like it's checking for specific rule violations and returning a specified error const?

Comment (adrienntindall) [reply src/battle_regulation.c:89]: also a0 -> ruleset

File: src/battle_regulation.c [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+        for (i = 0; i < PARTY_SIZE - 1; ++i) {
+            for (j = i + 1; j < PARTY_SIZE; ++j) {
+                if (heldItems[i] == heldItems[j] && species[i] != SPECIES_NONE && heldItems[i] != ITEM_NONE) {
+                    return 3;
+                }
+            }
+        }
+    }
+
+    return 0;
```
Comment (adrienntindall) [inline src/battle_regulation.c:175]: RULESET_ERROR_NONE
^error consts like this perhaps

**Review by adrienntindall — CHANGES_REQUESTED** [review]

File: src/battle_regulation.c (hunk 3 above)
Comment (PikalaxALT) [reply src/battle_regulation.c:15]: `Pokedex` is already used

File: src/battle_regulation.c (hunk 4 above)
Comment (PikalaxALT) [reply src/battle_regulation.c:31]: Handled in #332

File: src/battle_regulation.c (hunk 5 above)
Comment (PikalaxALT) [reply src/battle_regulation.c:89]: Handled in #332

File: src/battle_regulation.c (hunk 6 above)
Comment (PikalaxALT) [reply src/battle_regulation.c:175]: Will handle in #332

Comment (PikalaxALT) [reply src/battle_regulation.c:175]: jk will handle now :)

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**